packages/protocol/contracts/common/LibraryProxy.sol

```diff
@@ -0,0 +1,86 @@
+pragma solidity ^0.5.13;
+/* solhint-disable no-inline-assembly, no-complex-fallback, avoid-low-level-calls */
+
+import "openzeppelin-solidity/contracts/utils/Address.sol";
+import "./Registry.sol";
+
+/**
+ * @title A library proxy utilizing the Unstructured Storage pattern, allowing
+ * for calls to a linked library pointed to by a Registry.
+ * @dev This contract should be inherited from (e.g. for a library called Math,
+ * create `contract MathProxy is LibraryProxy`). The inheriting contract should
+ * implement a fallback function that calls `_delegateToRegisteredLibrary`,
+ * passing in a Registry hashed identifier referencing the implementation
+ * library.
+ */
+contract LibraryProxy {
+  // Used to store the address of the registry contract.
+  // NOTE: non-standard EIP1967 value to not clash with the calling contract's
+  // underlying Proxy.
+  bytes32 private constant REGISTRY_POSITION = bytes32(
+    uint256(keccak256("eip1967.proxy.registry")) - 1
+  );
+
+  function _setRegistry(address registryAddress) public {
```

Comment:
why is this public?

Reply:
It definitely *can* safely be public - LibraryProxy uses the original contract's storage, we don't care about the storage at the address where a LibraryProxy ends up deployed at.

With the shim approach, would this have even been callable by the calling contract if this were `internal`? Currently, a `delegatecall` attempting to call <this function's signature> would have been issued to this contract's deployed address, would this be reachable by that calling convention? 

Either way, will make this `internal` if we go with the inheritance approach or remove altogether if we go with hardcoded Registry address.

Reply:
I agree that it's safe to make public, but it feels out of place. I don't have a particular reason for preferring it to be non-public, other than cleanliness